client/src/main/java/se/fortnox/reactivewizard/client/HttpClient.java

```diff
@@ -590,6 +590,24 @@ protected String getPath(Method method, Object[] arguments, JaxRsMeta meta) {
         return result;
     }
 
+    /**
+     * Recursive function getting all declared fields from the passed in class and its ancestors
+     *
+     * @param clazz the clazz
+     * @param fields the fields found so far.
+     *
+     * @return list of fields
+     */
+    private List<Field> getDeclaredFieldsFromClassAndAncestors(Class clazz, List<Field> fields) {
+        fields.addAll(asList(clazz.getDeclaredFields()));
+
+        if (clazz.getSuperclass() != null && !Object.class.equals(clazz.getSuperclass())) {
```

Comment:
Usually the terminal condition is placed in the top, when doing recursion. In this cade it would make sens. We would then not need the negation inside the if?

A field may be declared twice? In a subclass and in the patent? Maybe a set would be better than a list?

Reply:
A set wouldn't change anything since the Field object is tied to its declaring class and a Set would still contain all fields.
And if they were the same a Field in the parent class could have the @QueryParam annotation and the field in the subclass  not so it is desirable to get all fields.